File: src/main/java/io/zeitwert/ddd/search/service/api/SearchService.java

package io.zeitwert.ddd.search.service.api;

import java.util.List;

import io.zeitwert.ddd.aggregate.model.Aggregate;
import io.zeitwert.ddd.search.model.SearchResult;

public interface SearchService {

	/**
	 * Store the search texts and tokens.
	 *
	 * @param aggregate aggregate to store
	 * @param texts     list of texts to be stored
	 * @param tokens    list of tokens to be stored
	 */
	void storeSearch(Aggregate aggregate, List<String> texts, List<String> tokens);

	List<SearchResult> find(String searchText, int maxResultSize);

	List<SearchResult> find(List<String> itemTypes, String searchText, int maxResultSize);

}

File: fm-domain/src/main/java/io/zeitwert/app/ddd/service/api/SearchService.kt
package io.zeitwert.app.ddd.service.api

import dddrive.ddd.model.Aggregate
import io.zeitwert.app.ddd.model.SearchResult
import io.zeitwert.app.session.model.SessionContext

interface SearchService {

	/**
	 * Store the search texts and tokens.
	 *
	 * @param aggregate aggregate to store
	 * @param texts     list of texts to be stored
	 * @param tokens    list of tokens to be stored
	 */
	fun storeSearch(
		aggregate: Aggregate,
		texts: List<String?>,
		tokens: List<String?>,
	)

	fun findOne(
		sessionContext: SessionContext,
		itemType: String,
		searchText: String,
	): SearchResult?

	fun find(
		sessionContext: SessionContext,
		searchText: String,
		maxResultSize: Int,
	): List<SearchResult>

	fun find(
		sessionContext: SessionContext,
		itemTypes: List<String>?,
		searchText: String,
		maxResultSize: Int,
	): List<SearchResult>

}
